File: backend/src/main/java/com/kylinops/common/enums/IntentType.java
package com.kylinops.common.enums;

/**
 * Agent 意图分类
 * <p>
 * IntentClassifier 将用户自然语言输入解析为以下意图之一，
 * 后续 ToolPlanningService 根据意图编排对应工具链。
 * </p>
 */
public enum IntentType {

    /** 系统健康检查 — 多 Tool 发散获取系统状态 */
    SYSTEM_CHECK,

    /** 磁盘诊断 — 磁盘用量分析和清理建议 */
    DISK_DIAGNOSIS,

    /** 服务诊断 — 检查服务运行状态 */
    SERVICE_DIAGNOSIS,

    /** 进程查询 — 查看进程列表或详细信息 */
    PROCESS_QUERY,

    /** 文件操作 — 文件清理、日志截断等 */
    FILE_OPERATION,

    /** 命令执行 — 执行运维命令（高度警惕） */
    COMMAND_EXECUTION,

    /** 网络查询 — 端口状态、网络连接 */
    NETWORK_QUERY,

    /** 日志查询 — 查看系统或应用日志 */
    LOG_QUERY,

    /** 通用对话 — 非运维类闲聊 */
    GENERAL_CHAT,

    /** 无法识别的意图 */
    UNKNOWN
}

File: frontend/src/utils/intentType.ts
// IntentType 兼容映射（前后端枚举不一致 → 统一到后端命名）。
// 不修改任何现有枚举值；只做兼容层。

const LEGACY_TO_CANONICAL: Readonly<Record<string, string>> = {
  HEALTH_CHECK: 'SYSTEM_CHECK',
  PROCESS_INQUIRY: 'PROCESS_QUERY',
  NETWORK_INQUIRY: 'NETWORK_QUERY',
  LOG_INQUIRY: 'LOG_QUERY',
  SERVICE_OPERATION: 'SERVICE_DIAGNOSIS',
};

export function normalizeIntentType(raw: string | undefined): string {
  if (!raw) return 'UNKNOWN';
  return LEGACY_TO_CANONICAL[raw] ?? raw;
}

const RCA_TITLES: Readonly<Record<string, string>> = {
  DISK_DIAGNOSIS: '根因分析链',
  SYSTEM_CHECK: '健康评估链',
  SERVICE_DIAGNOSIS: '服务诊断链',
  COMMAND_EXECUTION: '安全决策链',
};

export function rcaTitleFor(rawIntent: string | undefined): string | null {
  const canonical = normalizeIntentType(rawIntent);
  return RCA_TITLES[canonical] ?? null;
}